grakn/rpc/session.py

```diff
@@ -47,7 +50,9 @@ def __init__(self, client, database: str, session_type: SessionType, options=Gra
 
         self._session_id = self._grpc_stub.session_open(open_req).session_id
         self._is_open = True
-        self._pulse = self._scheduler.enter(5, 1, self._transmit_pulse, ())
+        self._scheduler.enter(5, 1, self._transmit_pulse, ())
+        thread = Thread(target=self._scheduler.run)
+        thread.start()
```

Comment:
The reason pulses were never triggered was because we were never calling `self._scheduler.run`.